src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -799,12 +799,25 @@ def test_children_is_orphan_validation():
     # It can be added when it has been detached from its previous parent
     schedule2.addchild(statement.detach())
 
-    # The only exception at the moment is if it is already the same
-    # parent. This is because the parent-child is currently set in 2 steps,
-    # and the parent reference may already been provided, but is causes false
-    # negatives like the case below.
-    # TODO: #294 Could solve this issue by making the parent-child set atomic
-    schedule2.addchild(schedule2.children[0])
+
+@pytest.mark.xfail(reason="Adding non-orphan nodes to the same parent where "
+                          "it already belongs to doesn't make it fail at the "
+                          "moment. #294. Could solve this issue by making the"
+                          " parent-child set atomic.")
+def test_children_is_orphan_same_parent():
+    ''' Test children addition operations with a node that is not an orphan
+    but is belongs to the same parent where it is added to.'''
+    # Create 2 schedules and add a child in the first Schedule
+    schedule1 = Schedule()
+    schedule2 = Schedule()
+    statement = Return(parent=schedule1)
+    schedule1.addchild(statement)
+
+    with pytest.raises(GenerationError) as error:
+        schedule2.addchild(schedule2.children[0])
```

Comment:
This should be:

    `schedule1.addchild(schedule1[0])`

I think and then `schedule2` can be removed?

Reply:
Done